src/main/java/twilightforest/util/WorldUtil.java

```diff
@@ -62,6 +63,23 @@ public static BlockPos randomOffset(RandomSource random, BlockPos pos, int rx, i
 		return pos.offset(dx, dy, dz);
 	}
 
+	public static <T> T getRandomElement(List<T> list, RandomSource rng) {
+		return list.get(rng.nextInt(list.size()));
+	}
+
+	public static <T> T getRandomElementWithWeights(List<Pair<T, Integer>> list, RandomSource rng) {
+		int totalWeight = list.stream().mapToInt(Pair::getSecond).sum();
+		int randomValue = rng.nextInt(totalWeight);
+
+		for (Pair<T, Integer> pair : list) {
+			randomValue -= pair.getSecond();
+			if (randomValue < 0) {
+				return pair.getFirst();
+			}
+		}
+		return list.get(rng.nextInt(list.size())).getFirst(); // This line should never be reached if input list is valid
```

Comment:
can this be replaced with `return getRandomElemen(list, rng).getFirst()` to reduce duplicate code?